cpp/src/io/comp/uncomp.cpp

```diff
@@ -219,7 +220,7 @@ int cpu_inflate(uint8_t *uncomp_data, size_t *destLen, const uint8_t *comp_data,
   z_stream strm;
 
   memset(&strm, 0, sizeof(strm));
-  strm.next_in   = (Bytef *)comp_data;
+  strm.next_in   = const_cast<Bytef *>(reinterpret_cast<Bytef const *>(comp_data));
```

Comment:
Wow, a very smelly code hidden behind a cast here. 
Defining ZLIB_CONST would make `next_in` const, but I don't know if it affects zlib performance. I think we can instead pass a non-const `comp_data` pointer here (note for a separate PR).

Reply:
Smelly indeed.
I intentionally left it like that as we have to account for both the scenario where `ZLIB_CONST` is defined as `const` and not.
The other contexts where the function `cpu_inflate` is used requires that it is `const` (i.e. they pass `const` arguments). So we'd have to perform the `const_cast` anyway.
I think the best solution would be to remove `Z_CONST`, but I'm not sure how that'd affect the output and performance as it'd require doing more investigation as I'd rather not make assumptions.